src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp

```diff
@@ -146,16 +146,18 @@ void ReadBufferFromRabbitMQConsumer::initExchange()
      * in current case we use hash exchange for binding to another exchange of some other type, which needs its own routing keys
      * of other types: headers, patterns and string-keys. This means that hash property must be changed.
      */
-    AMQP::Table binding_arguments;
-    binding_arguments["hash-property"] = "message_id";
-
-    /// Declare exchange for sharding.
-    consumer_channel->declareExchange(local_hash_exchange, AMQP::consistent_hash, binding_arguments)
-    .onError([&](const char * message)
     {
-        local_exchange_declared = false;
-        LOG_ERROR(log, "Failed to declare {} exchange: {}", exchange_type, message);
-    });
+        AMQP::Table binding_arguments;
```

Comment:
Added scope just to avoid capturing a lot of variables in lambda?

Reply:
-Wshadow